src/main/java/net/minecraftforge/common/brewing/IBrewingRecipe.java

```diff
@@ -42,4 +48,31 @@
      * ingredient. Empty if invalid input or ingredient.
      */
     ItemStack getOutput(ItemStack input, ItemStack ingredient);
+
+    @Override
+    default boolean matches(Container container, Level level)
+    {
+        if (container instanceof BrewingStandBlockEntity)
+            return (isInput(container.getItem(0)) || isInput(container.getItem(1)) || isInput(container.getItem(2))) && isIngredient(container.getItem(3));
+        else
+            return isInput(container.getItem(0)) && isInput(container.getItem(1));
+    }
+
+    @Override
+    default ItemStack assemble(Container container)
+    {
+        return getOutput(container.getItem(0), container.getItem(1));
```

Comment:
If the brewing stand has a special matching scenario, why doesn't assemble? Also, what is the point of the deferred method call?

Reply:
the method needs to be implemented but is actually never used so it can work in non vanilla brewing stands too